localstack/aws/handlers/service.py

```diff
@@ -63,17 +57,8 @@ def __call__(self, chain: HandlerChain, context: RequestContext, response: Respo
 
         return self.parse_and_enrich(context)
 
-    def get_parser(self, service: ServiceModel):
-        name = service.service_name
-
-        if name in self.parsers:
-            return self.parsers[name]
-
-        self.parsers[name] = create_parser(service)
-        return self.parsers[name]
-
     def parse_and_enrich(self, context: RequestContext):
-        parser = self.get_parser(context.service)
+        parser = create_parser(context.service)
```

Comment:
i guess it should be fine creating the parser every time! very little overhead since it caries no state.